mcp/runtime/src/main/java/io/quarkiverse/langchain4j/mcp/runtime/McpClientHealthCheck.java

```diff
@@ -0,0 +1,60 @@
+package io.quarkiverse.langchain4j.mcp.runtime;
+
+import java.lang.annotation.Annotation;
+import java.util.HashMap;
+import java.util.Map;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Any;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+import org.eclipse.microprofile.health.Readiness;
+
+import dev.langchain4j.mcp.client.McpClient;
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.InjectableBean;
+import io.quarkus.arc.InstanceHandle;
+
+@ApplicationScoped
+@Readiness
+public class McpClientHealthCheck implements HealthCheck {
+
+    private Map<String, McpClient> clientMap;
+
+    public McpClientHealthCheck() {
+        clientMap = getClientMap();
+    }
+
+    @Override
+    public HealthCheckResponse call() {
+        HealthCheckResponseBuilder builder = HealthCheckResponse.builder()
+                .name("MCP clients health check")
+                .up();
+        for (String name : clientMap.keySet()) {
+            McpClient client = clientMap.get(name);
+            try {
+                client.checkHealth();
+                builder.withData(name, "OK");
+            } catch (Exception e) {
+                builder.down().withData(name, e.getMessage());
+            }
+        }
+        return builder.build();
+    }
+
+    private Map<String, McpClient> getClientMap() {
+        Map<String, McpClient> map = new HashMap<>();
+        for (InstanceHandle<McpClient> handle : Arc.container().select(McpClient.class, Any.Literal.INSTANCE).handles()) {
```

Comment:
Minor gripe: Can't we make this a proper injection point instead of doing a programmatic lookup?

Reply:
Only if we merge https://github.com/quarkiverse/quarkus-langchain4j/pull/1393 first - I did here exactly what you [suggested](https://github.com/quarkiverse/quarkus-langchain4j/pull/1393#issuecomment-2765165445) the user should do instead :)

Reply:
kk, makes sense :)

Reply:
We need to access the name of the client.